src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -616,8 +711,18 @@ public CloudApplication getApplication(UUID applicationGuid) {
 
     @Override
     public InstancesInfo getApplicationInstances(String applicationName) {
-        CloudApplication application = getApplication(applicationName);
-        return getApplicationInstances(application);
+        UUID applicationGuid = getRequiredApplicationGuid(applicationName);
+        ApplicationState applicationState = getApplicationResponseByName(applicationName).getState();
+        return applicationState == ApplicationState.STARTED ? findApplicationInstances(applicationGuid) : null;
```

Comment:
Is returning null okay here? Is a `null` InstanceInfo handled gracefully somewhere down the line?

Reply:
It was handled but these checks were redundant at all